CRM/Contactlayout/BAO/ContactLayout.php

```diff
@@ -520,17 +520,25 @@ public static function getAllTabs() {
       'caller' => 'ContactLayout',
     ];
     CRM_Utils_Hook::tabset('civicrm/contact/view', $tabs, $context);
-    $allTabs = [];
-    foreach ($tabs as $index => $tab) {
-      // Every tab OUGHT to have an 'id' but the documentation about this has been unclear.
-      // Proactively convert array key to id if missing.
-      $allTabs[] = $tab + [
-        'is_active' => TRUE,
-        'id' => $index,
-      ];
+
+    if (method_exists(\CRM_Core_Smarty::class, 'setRequiredTabTemplateKeys')) {
```

Comment:
This checks if the method exists but not if the class exists. I guess `CRM_Core_Smarty` is going to be around for a long time yet but who knows?

Reply:
I wanted to check for the method because it's new, so prevents breaking sites older than 5.82. Wasn't thinking forward-compatibility ?